### PR TITLE
Tracking changes for `nucypher-core@0.11.0` update

### PR DESCRIPTION
**Pull request by piotr-roslaniec**
**Type of PR:**
- Other

**Required reviews:** 
- 2

**What this does:**
- Ensures compatibility with an upcoming `nucypher-core@0.11.0` release

**Issues fixed/closed:**
- Related to: https://github.com/nucypher/nucypher-core/pull/69

**Why it's needed:**
> Explain how this PR fits in the greater context of the NuCypher Network.
> E.g., if this PR address a `nucypher/productdev` issue, let reviewers know!

**Notes for reviewers:**
- Awaits release of `nucypher-core@0.11.0` and replacing the git dependencies to an actual Python package

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -11,11 +11,11 @@ python_version = "3"
 constant-sorrow = ">=0.1.0a9"
 bytestring-splitter = ">=2.4.0"
 hendrix = ">=4.0"
-nucypher-core = ">=0.10.0"
+nucypher-core = {git="https://github.com/nucypher/nucypher-core.git", ref="7f3a7f21f2f9bdc0395e256025ec9af0d53723aa", subdirectory="nucypher-core-python"}
```
Comment (derekpierre) [inline Pipfile:14]: Just a reminder to update this before merging, once the new version of `nucypher-core` is released.

File: nucypher/crypto/ferveo/dkg.py [hunk 2]
```diff
@@ -19,14 +19,11 @@
...
 
-class FerveoVariant(Enum):
-    SIMPLE = 0
-    PRECOMPUTED = 1
-
-
 _VARIANTS = {
-    FerveoVariant.SIMPLE: AggregatedTranscript.create_decryption_share_simple,
-    FerveoVariant.PRECOMPUTED: AggregatedTranscript.create_decryption_share_precomputed
+    str(FerveoVariant.simple): AggregatedTranscript.create_decryption_share_simple,
```
Comment (derekpierre) [inline nucypher/crypto/ferveo/dkg.py:23]: Do we have to key by string? We can't use the enum instead?

**Review by derekpierre — APPROVED** [review]
🎸 Looks great.

One outstanding little nitpick (😅 ) in the `ferveo` linked PR about the naming of the enums `simple`/`precomputed`.

File: nucypher/crypto/ferveo/dkg.py (hunk 2 above)
Comment (piotr-roslaniec) [reply nucypher/crypto/ferveo/dkg.py]: :heavy_check_mark:

File: nucypher/characters/lawful.py [hunk 3]
```diff
@@ -712,7 +716,7 @@ def threshold_decrypt(
         conditions: Lingo,
         context: Optional[dict] = None,
         ursulas: Optional[List["Ursula"]] = None,
-        variant: str = "simple",
+        variant: FerveoVariant = FerveoVariant.Simple,
```
Comment (KPrasch) [inline nucypher/characters/lawful.py:719]: Nice to see this finally formalized.

**Review by KPrasch — APPROVED** [review]
🤠

**Comment by piotr-roslaniec** [comment]
I forgot about this part:
> Awaits release of nucypher-core@0.11.0 and replacing the git dependencies to an actual Python package

Will follow up and replace the git dependency shortly.

**Pull request merged**